dotcom-rendering/src/components/Discussion.tsx

```diff
@@ -374,6 +379,23 @@ const reducer = (state: State, action: Action): State => {
 				},
 			};
 		}
+		case 'expandCommentReplies': {
+			const replaceMatchingComment = (
```

Comment:
Maybe

```suggestion
			const findMatchingComment = (
```
?

Not sure what we are replacing with what here..

Reply:
This is a method that will recursively look for a comment with a matching ID and replace its responses with a new list.

```
━━ A
━┳ B
 ┣━ B.1
 ┣━ B.2
 ┡━ B.3
 ┆  (show more)
━┳ C
 ┣━ C.1
 ┗━ C.2
━┳ D
 ┣┳ D.1
 ┃┣━ D.1.A
 ┃┣━ D.1.B
 ┃┡━ D.1.C
 ┃┆ (show more)
 ┣━ D.2
 ┗━ D.3
```

If we get the whole list of responses for **B**, then we need to only replace **B**’s reponses. Similarly, if we get the whole list of responses for **D.1**, we want to replace all of **D.1**’s responses, but leave everything intact.

Reply:
Thanks for the documentation and the test!